frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -2651,7 +2756,12 @@ def tree_add(self, resulting_type: ConversionType, xs: List[Conversion]) -> Conv
             a = xs.pop()
             b = xs.pop()
 
-            result = self.add(resulting_type, a, b)
+            intermediate_type = self.tensor(
+                (self.eint if resulting_type.is_unsigned else self.esint)(resulting_type.bit_width),
```

Comment:
does not depend on `x should be done before the while for clarity.
`resulting_mlir_type`

Reply:
Done.